tests/rust/tcp-test/close/wait.rs

```diff
@@ -169,7 +169,7 @@ fn wait_after_async_close_connecting_socket(libos: &mut LibOS, remote: &SocketAd
 
     // Poll again to check that the queue token returns an err.
     match libos.wait(qt, Some(Duration::from_micros(0))) {
-        Ok(qr) if qr.qr_opcode == demi_opcode_t::DEMI_OPC_FAILED => {},
+        Ok(qr) if qr.qr_opcode == demi_opcode_t::DEMI_OPC_FAILED && qr.qr_ret == libc::ECANCELED => {},
```

Comment:
@ppenna What is the expected behavior here? The error returned is ECONNREFUSED because the second call to wait causes the connect to actually try to connect to the remote address. We don't want to overwrite that value on close/cancel because the operation completed before close was called and the error code might be needed.

Reply:
Well, it looks like that this test is not exercising the "wait_after_async_close_connecting_socket"